src/main/java/com/bokov/FirstTGBot/services/BTCService.java

```diff
@@ -0,0 +1,46 @@
+package com.bokov.FirstTGBot.services;
+
+import com.bokov.FirstTGBot.models.BTC;
+import com.bokov.FirstTGBot.repositores.BTCRepository;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.RestTemplate;
+
+
+import java.util.Date;
+import java.util.List;
+
+
+@Service
+@Transactional(readOnly = true)
+public class BTCService {
+    private final BTCRepository btcRepository;
+    @Autowired
+    private Environment environment;
+
+
+    @Autowired
```

Comment:
@Autowired annotation is most using in integrations tests and abstract classes where you can not pass bean through constructor

you can get rid of constructor and code might look like:

@Service
@Transactional(readOnly = true)
@RequiredArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
public class BTCService {

    BTCRepository btcRepository;
    Environment environment;